gems/aws-sdk-core/lib/seahorse/client/plugins/response_target.rb

```diff
@@ -41,7 +41,9 @@ def add_event_listeners(context, target)
             context.http_response.on_error do
               body = context.http_response.body
               File.unlink(body) if ManagedFile === body
-              context.http_response.body = StringIO.new
+              unless context.http_response.body.respond_to?(:io)
```

Comment:
It may be worth adding a comment explaining why this conditional is important for later reference if looking at this code months later.